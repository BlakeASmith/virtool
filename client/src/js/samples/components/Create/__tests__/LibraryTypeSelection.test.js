import { LibraryTypeSelection } from "../LibraryTypeSelection";

describe("<LibraryTypeSelection>", () => {
    let props;
    beforeEach(() => {
        props = {
            onSelect: jest.fn(),
            libraryType: "normal"
        };
    });

    it("should select Normal type when [LibraryTypeSelection='normal']", () => {
        const wrapper = shallow(<LibraryTypeSelection {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should select sRNA type when [LibraryTypeSelection='srna']", () => {
        props.libraryType = "srna";
        const wrapper = shallow(<LibraryTypeSelection {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
    it("should select Amplicon type when [LibraryTypeSelection='amplicon']", () => {
        props.libraryType = "amplicon";
        const wrapper = shallow(<LibraryTypeSelection {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
    it("should call onSelect with libraryType normal when Normal type is clicked", () => {
        const wrapper = shallow(<LibraryTypeSelection {...props} />);
        wrapper
            .find("SelectBox")
            .at(0)
            .simulate("click");
        expect(props.onSelect).toHaveBeenCalledWith("normal");
    });
    it("should call onSelect with libraryType srna when srna type is clicked", () => {
        const wrapper = shallow(<LibraryTypeSelection {...props} />);
        wrapper
            .find("SelectBox")
            .at(1)
            .simulate("click");
        expect(props.onSelect).toHaveBeenCalledWith("srna");
    });
    it("should call onSelect with libraryType amplicon when amplicon type is clicked", () => {
        const wrapper = shallow(<LibraryTypeSelection {...props} />);
        wrapper
            .find("SelectBox")
            .at(2)
            .simulate("click");
        expect(props.onSelect).toHaveBeenCalledWith("amplicon");
    });
});
