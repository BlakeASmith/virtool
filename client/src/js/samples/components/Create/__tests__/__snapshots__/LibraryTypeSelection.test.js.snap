// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LibraryTypeSelection> should select Amplicon type when [LibraryTypeSelection='amplicon'] 1`] = `
<Fragment>
  <label>
    Library Type
  </label>
  <LibraryTypeSelection__LibraryTypeSelectBoxContainer>
    <SelectBox
      active={false}
      onClick={[Function]}
    >
      <div>
        Normal
      </div>
      <span>
        Search against whole genome references using normal reads.
      </span>
    </SelectBox>
    <SelectBox
      active={false}
      onClick={[Function]}
    >
      <div>
        sRNA
      </div>
      <span>
        Search against whole genome references using sRNA reads
      </span>
    </SelectBox>
    <SelectBox
      active={true}
      onClick={[Function]}
    >
      <div>
        Amplicon
      </div>
      <span>
        Search against barcode references using amplicon reads.
      </span>
    </SelectBox>
  </LibraryTypeSelection__LibraryTypeSelectBoxContainer>
</Fragment>
`;

exports[`<LibraryTypeSelection> should select Normal type when [LibraryTypeSelection='normal'] 1`] = `
<Fragment>
  <label>
    Library Type
  </label>
  <LibraryTypeSelection__LibraryTypeSelectBoxContainer>
    <SelectBox
      active={true}
      onClick={[Function]}
    >
      <div>
        Normal
      </div>
      <span>
        Search against whole genome references using normal reads.
      </span>
    </SelectBox>
    <SelectBox
      active={false}
      onClick={[Function]}
    >
      <div>
        sRNA
      </div>
      <span>
        Search against whole genome references using sRNA reads
      </span>
    </SelectBox>
    <SelectBox
      active={false}
      onClick={[Function]}
    >
      <div>
        Amplicon
      </div>
      <span>
        Search against barcode references using amplicon reads.
      </span>
    </SelectBox>
  </LibraryTypeSelection__LibraryTypeSelectBoxContainer>
</Fragment>
`;

exports[`<LibraryTypeSelection> should select sRNA type when [LibraryTypeSelection='srna'] 1`] = `
<Fragment>
  <label>
    Library Type
  </label>
  <LibraryTypeSelection__LibraryTypeSelectBoxContainer>
    <SelectBox
      active={false}
      onClick={[Function]}
    >
      <div>
        Normal
      </div>
      <span>
        Search against whole genome references using normal reads.
      </span>
    </SelectBox>
    <SelectBox
      active={true}
      onClick={[Function]}
    >
      <div>
        sRNA
      </div>
      <span>
        Search against whole genome references using sRNA reads
      </span>
    </SelectBox>
    <SelectBox
      active={false}
      onClick={[Function]}
    >
      <div>
        Amplicon
      </div>
      <span>
        Search against barcode references using amplicon reads.
      </span>
    </SelectBox>
  </LibraryTypeSelection__LibraryTypeSelectBoxContainer>
</Fragment>
`;
